TSE Review - rspec

carlosveigadev opened the pull request:
- We created specified tests using rspec gem for the following methods:
```
#my_each;
#my_each_with_index;
#my_select;
#my_all?;
#my_any;
#my_none;
#my_count;
#my_map;
#my_inject;
```

carlosveigadev committed:
create rspec files

carlosveigadev committed:
add iteration over words for my_each test

add --format documentation to .rspec

carlosveigadev committed:
add test for iteration of numbers with my_each

carlosveigadev committed:
add test to return Enumerator when a block is not passed to my_each

carlosveigadev committed:
add test for array of words to my_each_with_index

add test for array of numbers to my_each_with_index

carlosveigadev committed:
add my_each_with_index test to return indices of iteration over words

carlosveigadev committed:
add test to return Enumerator when block is not passed to my_each_with_index

carlosveigadev committed:
fix rubocop offenses

jacobo12montes committed:
make the test for my_select method

jacobo12montes committed:
make the test for my_all method

jacobo12montes committed:
make the test for my_any method

carlosveigadev committed:
add tests to verify if my_none? works properly, with array, numbers, words, class and falsy values

carlosveigadev committed:
fix rubocop offenses

carlosveigadev committed:
add tests for my_count, with no arguments, with arguments and with a block passed

carlosveigadev committed:
add tests to my_map, when a block is given and if it works as desired

jacobo12montes committed:
make the test for my_inject method

carlosveigadev committed:
fix my_each_with_index test

fix rubocop offenses

lekegitrepo commented on test.rb:
```diff
@@ -53,23 +54,24 @@
 ary = [1, 2, 4, 2]
-puts ary.count               #=> 4
-puts ary.count(2)            #=> 2
-puts ary.count{ |x| x%2==0 } #=> 3
+puts ary.my_count               #=> 4
+puts ary.my_count(2)            #=> 2
+puts ary.my_count{ |x| x%2==0 } #=> 3
```
**_#my_count counts the number of items in enum that are equal to an item if an argument is given_**

lekegitrepo commented on test.rb:
```diff
@@ -53,23 +54,24 @@
 puts 'my_inject'
-puts (5..10).my_inject { |sum, n| sum + n }         #=> 45
-puts (5..10).inject(1) { |product, n| product * n } #=> 151200
+puts (5..10).my_inject(2) { |sum, n| sum + n }         #=> 45
+puts (5..10).my_inject(2) { |product, n| product * n } #=> 151200
```
**_#my_inject should raise a "LocalJumpError" when no block or argument is given_**

lekegitrepo reviewed (changes requested):
### Status: Request changes :recycle: 
### Hi @wrakc and @jacobo12montes :clap: :clap:
#### You have done a good job :+1: However two test cases are still failing.

**_Remove files that are not related (.vscode)_** 

**_Fix these and submit them for review._**

#### Check my comments below about the failing test cases.

#### _Reviewed by Temitayo Adeleke._

carlosveigadev committed:
remove not necessary files

carlosveigadev committed:
fix my_inject to raise LocalJumpError when needed

add LocalJumpError test to my_inject on rspec

carlosveigadev committed:
remove unecessary files

fix rubocop offenses

carlosveigadev committed:
fix rubocop offenses

carlosveigadev commented:
@lekegitrepo thanks for the feedback, we improved the code as suggested.
- Removed unnecessary files;
- Added test to raise LocalJumpError to my_inject.

andresporras3423 commented on spec/main_spec.rb:
```diff
@@ -0,0 +1,202 @@
+      expect([].my_none?).to eql(true)
+    end
+
+    it 'when an array is nill, and no block is given, returns true because there is nothing to compare with array' do
+      expect([nil].my_none?).to eql(true)
+    end
+
+    it 'when an array is nill or false, and no block is given, should return true because there is nothing to compare with the array' do
```
replace "nill" with "nil"

andresporras3423 commented on spec/main_spec.rb:
```diff
@@ -0,0 +1,202 @@
+
+  describe '#my_all' do
+    it 'should return true if all the elements on the argument return true after passing the block' do
+      expect(array_words.my_all? { |word| word.length >= 3 }).to eql(true)
```
Test some negative scenarios using to_not eq:
### Example
`expect(array_words.my_all? { |word| word.length >= 3 }).to eql(true)`
`expect(array_words.my_all? { |word| word.length >= 3 }).to_not eql(false)`

andresporras3423 reviewed (changes requested):
### Status: Changes Required :recycle:
**Hello @wrakc @jacobo12montes** 
Good job with your new submission for this project :muscle:  :muscle:
Your tests passes most of the requirements :clap: :clap: :clap:  
There are just a couple of details to improve before you can go ahead, please check out below the comments to your code.

Please request another review once you are ready! :+1: 

Happy coding :relaxed:

Oscar Andrés Russi Porras

jacobo12montes committed:
fix typos

jacobo12montes committed:
add some negative test to main_spec file

jacobo12montes commented:
@andresporras3423 we made the improvements requested:
-Fix some typos
-Add  negative tests to the spec file

andresporras3423 reviewed (changes requested):
### Status: Changes Required :recycle:
**Hello @wrakc @jacobo12montes** 
Great job following the improvements of the last review :muscle:  :muscle: 
We still need to fix one last detail:
- [ ] I have noticed you modified the rubocop default configuration. You should use the original configuration.

Please request another review once you are ready! :+1: 

Happy coding :relaxed:

Oscar Andrés Russi Porras

carlosveigadev committed:
add original rubocop config

carlosveigadev committed:
fix rubocop offenses

carlosveigadev commented:
Thanks, @andresporras3423, we improved the code as suggested:
- Added original rubocop .yml configuration.
- Fix rubocop offenses on rspecs files.

krys2fa commented:
# Status: Project Approved! 🥇 
Congratulations on making all the requested changes and getting this project approved. 👍 

You can now merge this branch into the master branch and go on with your next project.

💡  Do not forget to submit your project completion form after merging this branch.

All the best team,
Chris
